src/Raven.Server/Documents/Handlers/Admin/AdminIndexHandler.cs

```diff
@@ -21,14 +20,15 @@ public class AdminIndexHandler : DatabaseRequestHandler
         {
             using (ContextPool.AllocateOperationContext(out DocumentsOperationContext context))
             {
-                var createdIndexes = new List<string>();
+                var createdIndexes = new List<PutIndexResult>();
                 var input = await context.ReadForMemoryAsync(RequestBodyStream(), "Indexes");
                 if (input.TryGet("Indexes", out BlittableJsonReaderArray indexes) == false)
                     ThrowRequiredPropertyNameInRequest("Indexes");
 
                 foreach (var indexToAdd in indexes)
                 {
                     var indexDefinition = JsonDeserializationServer.IndexDefinition((BlittableJsonReaderObject)indexToAdd);
+                    indexDefinition.Name = indexDefinition.Name.Trim();
```

Comment:
This can throw NRE and name validation is done later in code.

Reply:
You can change that to: `indexDefinition.Name?.Trim();`

Reply:
ok